syft is now throwing panic with version 0.66.0

**Please provide a set of steps on how to reproduce the issue**

Run syft version 0.66.0 against an image in DockerHub now produces an error.

syft packages registry:xxxxx/xxxx:xxxxx --scope squashed -o table -v 

**What happened**:

panic: interface conversion: interface {} is *pe.OptionalHeader32, not *pe.OptionalHeader64

goroutine 243 [running]:
github.com/anchore/syft/syft/pkg/cataloger/java.newPE({0xc001c9bb00, 0x3c}, {0x7f4b44402c68, 0xc04d772de0?})
	/home/runner/work/syft/syft/syft/pkg/cataloger/java/graalvm_native_image_cataloger.go:234 +0x4e9
github.com/anchore/syft/syft/pkg/cataloger/java.fetchPkgs({0x7f4b44402c30?, 0xc04d772de0?}, {0xc001c9bb00, 0x3c})
	/home/runner/work/syft/syft/syft/pkg/cataloger/java/graalvm_native_image_cataloger.go:510 +0x219
github.com/anchore/syft/syft/pkg/cataloger/java.(*NativeImageCataloger).Catalog(0xc00007e360?, {0x145dd68, 0xc0077d80c8})
	/home/runner/work/syft/syft/syft/pkg/cataloger/java/graalvm_native_image_cataloger.go:544 +0x4e5
github.com/anchore/syft/syft/pkg/cataloger.runCataloger({0x1456150, 0x1d55e10}, {0x145dd68?, 0xc0077d80c8})
	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:57 +0x188
github.com/anchore/syft/syft/pkg/cataloger.Catalog.func1()
	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:127 +0x105
created by github.com/anchore/syft/syft/pkg/cataloger.Catalog
	/home/runner/work/syft/syft/syft/pkg/cataloger/catalog.go:122 +0x2aa


**What you expected to happen**:

Scan should proceed with results produced.

**Anything else we need to know?**:

**Environment**:
- Output of `syft version`: 0.66.0
- OS (e.g: `cat /etc/os-release` or similar): 22.04.1 LTS (Jammy Jellyfish)

--- a/syft/pkg/cataloger/java/graalvm_native_image_cataloger.go
+++ b/syft/pkg/cataloger/java/graalvm_native_image_cataloger.go
@@ -237,6 +237,8 @@ func newPE(filename string, r io.ReaderAt) (nativeImage, error) {
 		exportSymbolsDataDirectory = h.DataDirectory[0]
 	case *pe.OptionalHeader64:
 		exportSymbolsDataDirectory = h.DataDirectory[0]
+	default:
+		return nil, fmt.Errorf("unable to get exportSymbolsDataDirectory from binary: %s", filename)
 	}
 	// If we have no exported symbols it is not a Native Image
 	if exportSymbolsDataDirectory.Size == 0 {

--- a/syft/pkg/cataloger/java/graalvm_native_image_cataloger.go
+++ b/syft/pkg/cataloger/java/graalvm_native_image_cataloger.go
@@ -231,8 +231,13 @@ func newPE(filename string, r io.ReaderAt) (nativeImage, error) {
 	if err != nil {
 		return fileError(filename, err)
 	}
-	optionalHeader := bi.OptionalHeader.(*pe.OptionalHeader64)
-	exportSymbolsDataDirectory := optionalHeader.DataDirectory[0]
+	var exportSymbolsDataDirectory pe.DataDirectory
+	switch h := bi.OptionalHeader.(type) {
+	case *pe.OptionalHeader32:
+		exportSymbolsDataDirectory = h.DataDirectory[0]
+	case *pe.OptionalHeader64:
+		exportSymbolsDataDirectory = h.DataDirectory[0]
+	}
 	// If we have no exported symbols it is not a Native Image
 	if exportSymbolsDataDirectory.Size == 0 {
 		return fileError(filename, errors.New(nativeImageMissingExportedDataDirectoryError))

